src/install.c

```diff
@@ -343,19 +343,30 @@ static gboolean slot_list_contains(GList *slotlist, const RaucSlot *testslot)
 	return FALSE;
 }
 
-static RaucSlot* get_bootname_slot(GHashTable *targetgroup, GError **error)
+/* Get the slot that has a bootname set, i.e. that we need to manipulate by the
+ * boot selection layer to allow atomic updating.
+ */
+static RaucSlot* get_bootname_slot(GHashTable *targetgroup, GList *install_images, GError **error)
 {
-	GHashTableIter iter;
-	RaucSlot *iterslot = NULL;
 	RaucSlot *bootnameslot = NULL;
 
-	/* Sanity check: only one bootname allowed */
-	g_hash_table_iter_init(&iter, targetgroup);
-	while (g_hash_table_iter_next(&iter, NULL, (gpointer*) &iterslot)) {
+	/* Iterate over all slots we have an image to install for */
+	for (GList *l = install_images; l != NULL; l = l->next) {
+		RaucSlot *iterslot = g_hash_table_lookup(targetgroup, ((RaucImage*) l->data)->slotclass);
+
+		/* If our slot has a parent with bootname set, use this for
+		 * testing (setting both parent and bootname is not allowed) */
```

Comment:
I do not understand the supplementary explanation in parentheses - could you rephrase it a little bit?